src/compiler.ts

```diff
@@ -6224,6 +6215,25 @@ export class Compiler extends DiagnosticEmitter {
           this.currentType = tee ? local.type : Type.void;
           return module.unreachable();
         }
+        if (local.closureContextOffset > 0) {
+          var nativeType = valueType.toNativeType();
+          return module.block(null, [
+            module.store(
+              local.type.byteSize,
+              module.local_get(0, this.options.nativeSizeType),
+              valueExpr,
+              valueType.toNativeType(),
+              local.closureContextOffset
+            ),
+            module.load(
+              local.type.byteSize,
+              valueType.is(TypeFlags.SIGNED),
+              module.local_get(0, this.options.nativeSizeType),
+              valueType.toNativeType(),
+              local.closureContextOffset
+            )
```

Comment:
Store the value into the closure context, then re-load that value since `makeAssignment` expects the value to be tee'd and it doesn't seem like `module.store` can tee. (using a local instead is a possible optimization, but just wanted to get this up first because it's more clear to read)